highlight/highlight
# Add a way to track the referrer of error alerts

## Summary



This PR attaches a `referrer` query param onto error alert URLs. When the URL is visited, we detect for the presence of this and attach it to the mixpanel `Viewed Error` [event](https://mixpanel.com/project/2504464/view/3045720/app/events#VsdjWdksXZzA). 

![Screenshot_2023-07-18_at_2_46_52_PM](https://github.com/highlight/highlight/assets/58678/d6066d48-3fed-4b02-8446-b95551417108)

_Note: I only applied this logic to error alerts since it's what we're focused most on right now. Doing the other alerts would require some additional digging into how to integrate into the existing tracking events we have (or if we have to create brand new ones)._


## How did you test this change?



To test:
* Set up an error alert with slack, discord, email, webhook
* The link should have `referrer=slack|discord|email|webhook`
* The tracking property should show up in mixpanel

Confirmed slack
See above

Confirmed discord:
![Screenshot 2023-07-18 at 2 52 39 PM](https://github.com/highlight/highlight/assets/58678/6c2d3c30-4757-439f-8e28-b8f0b7e6ca5d)


Confirmed email:
![Screenshot 2023-07-18 at 2 50 05 PM](https://github.com/highlight/highlight/assets/58678/f018c228-3e5d-46d3-91a7-6ebf273a1c89)

Confirmed webhook:
Using this [ngrok guide](https://ngrok.com/docs/integrations/hostedhooks/webhooks/) for webhooks, confirmed the passed URL has the `referrer` property:
![Screenshot 2023-07-18 at 3 10 20 PM](https://github.com/highlight/highlight/assets/58678/468ab25a-ebe2-4d0d-a822-0adc76a8bd56)

I copied/pasted the URL and confirmed it shows up in mixpanel

![Screenshot 2023-07-18 at 3 14 35 PM](https://github.com/highlight/highlight/assets/58678/4d9c7bee-2713-4493-9771-844730d2feae)



## Are there any deployment considerations?



N/A
